src/ApiTestHost/Controllers/DelayController.cs

```diff
@@ -43,7 +45,7 @@ public DelayController(ILogger<DelayController>? logger)
             _logger.LogInformation("Delaying request to \"{0}\" by {1} milliseconds...", url, delay);
             await Task.Delay(delay);
 
-            var targetUri = new UriBuilder(url) { Query = HttpContext.Request.QueryString.Value }.Uri;
+            var targetUri = new UriBuilder(HttpUtility.UrlDecode(url)) { Query = HttpContext.Request.QueryString.Value }.Uri;
```

Comment:
OK then I should test it once again and find a way to be able to handle both cases

Reply:
I checked the Code and it is working fine if the url is not encoded (Since I am using UrlDecode it here and it won't change  the url if it is not Encoded)

Reply:
but wouldn't it fail for something like this?
`http://localhost:8426/delay/1000/https://my-service/api/blub?test=%2Ftest%2F`
because it will falsely decode the "%2F"?

Reply:
Hmm good point